### DISCUSSION TITLE
Full JWT example with local accounts please?

### DISCUSSION THREAD

**Discussion by pbaletkeman** [Neoteroi/BlackSheep#422] [General]
I see this post, https://github.com/Neoteroi/BlackSheep/discussions/419 by @amirongit @  and this post https://github.com/Neoteroi/BlackSheep/discussions/367 by @tyzhnenko so I think what I'd like isn't impossible.
I would like a step by step guide/tutorial on how I can setup JWT with accounts stored in my local database.
I'm still a new to Python (started programming with it in June 2023) and only found out about BlackSheep about late September 2023.

@amirongit did mention that they were able to get JWT working fine in their application.
I think that what @tyzhnenko  has created is probably the way to go, but the final solution still eludes me.
From what I can tell the documentation on the site only mentions using providers like Auth0, Active Directory, Okta, etc.
I'd like to do everything locally without the third party service.

**Comment by tyzhnenko** [comment]
Hey @pbaletkeman I've just discovered that I forgot to put AppAuthProvider class implementation into #367. Take a look. You just need to implement a storage class to get user data from a database for authentication.

Feel free to ask any questions.

**Comment by tyzhnenko** [comment]
Here's an example with OAuth2 Password provider – https://github.com/Neoteroi/BlackSheep-Examples/pull/9
